### DISCUSSION TITLE
Images from Amazon S3 (Or any website really)

### DISCUSSION THREAD

**Discussion by dvelopin** [mpdf/mpdf#1579] [General]
I downloaded the latest files from the repo a few minutes ago and still have the issue. This is executed from the web.

Essentially I have a PDF I am making which loads several images from an Amazon S3 bucket. So a simple <img src=""/> tag in the html.

I keep getting the following PHP error:

fopen(https://s3.us-east-2.amazonaws.com/link-to-image.jpeg): failed to open stream: no suitable wrapper could be found

The file is vendor/mpdf/mpdf/src/Image/ImageProcessor.php and line number 227.

Is there a better way to load an image from Amazon S3?

This issue also occurs with other websites so not specific to Amazon S3.

Here is example code:

```php
$stylesheet = '';
$stylesheet .= file_get_contents('vendor/bootstrap/css/bootstrap.min.css');
$stylesheet .= '
<style>
    body {

    }
    @page {
        size: 8.5in 11in;
        margin: 0px;
    }
    .col-xs-1, .col-sm-1, .col-md-1, .col-lg-1, .col-xs-2, .col-sm-2, .col-md-2, .col-lg-2, .col-xs-3, .col-sm-3, .col-md-3, .col-lg-3, .col-xs-4, .col-sm-4, .col-md-4, .col-lg-4, .col-xs-5, .col-sm-5, .col-md-5, .col-lg-5, .col-xs-6, .col-sm-6, .col-md-6, .col-lg-6, .col-xs-7, .col-sm-7, .col-md-7, .col-lg-7, .col-xs-8, .col-sm-8, .col-md-8, .col-lg-8, .col-xs-9, .col-sm-9, .col-md-9, .col-lg-9, .col-xs-10, .col-sm-10, .col-md-10, .col-lg-10, .col-xs-11, .col-sm-11, .col-md-11, .col-lg-11, .col-xs-12, .col-sm-12, .col-md-12, .col-lg-12 {
        border:0;
        padding:0;
        margin-left:-0.00001;
        font-size: 10px;
    }
    .table-title {
        line-height: 10px;
        background: #CCC;
        font-weight: bold;
        padding: 8px;
        font-size: 10px;
    }
    .table-row {
        padding: 8px;
        font-size: 10px;
    }
</style>
';

$html = '
    <div class="container" style="padding: 25px 35px;">
        <div class="row">
            <div class="col-xs-12" style="background: #333333;">
                <h3 style="color: #FFF; padding-left: 10px;">Google\'s Logo</h3>
            </div>
        </div>
        <div class="row table-row">
            <div class="col-xs-12">
                <img src="https://www.google.com/logos/doodles/2021/seasonal-holidays-2021-6753651837109324.3-ladc.gif">
            </div>
        </div>
    </div>
';

// reference the Dompdf namespace
$mpdf = new \Mpdf\Mpdf(['format' => 'Letter']);
$mpdf->WriteHTML($stylesheet, 1);
$mpdf->WriteHTML($html, 2);
$mpdf->Output('Test.pdf', 'F');
```

The bootstrap CSS file being included here is a Bootstrap 3 CSS file.

The image being loaded here in this case is Google's current logo on their site.

And the error I get is:

```
fopen(https://www.google.com/logos/doodles/2021/seasonal-holidays-2021-6753651837109324.3-ladc.gif): failed to open stream: no suitable wrapper could be found
```

**Comment by dvelopin** [comment]
It keeps changing my HTML to actual HTML in the discussion. So attaching a screenshot. 

<img width="924" alt="Screen Shot 2021-12-13 at 1 01 36 PM" src="https://user-images.githubusercontent.com/10199166/145888235-4fcf6214-60b5-424e-abd1-f55b4ec97247.png">

**Comment by jakejackson1** [comment]
For loading images from S3: http://mpdf.github.io/reference/mpdf-variables/whitelist-stream-wrappers.html

**Reply by finwe** [reply]
https is present there by default and not altered in the source code. Also, the error message would be different, this is internal PHP message.

**Comment by finwe** [comment]
What does **Registered PHP Streams** in your PHP info say? Is https present? I guess https is missing?

**Reply by finwe** [reply]
Do you have OpenSSL extension installed and enabled?

**Reply by finwe** [reply]
What are your PHP installation details generally?

**Reply by dvelopin** [reply]
Registered PHP Streams | https, ftps, compress.zlib, php, file, glob, data, http, ftp, compress.bzip2, phar, zip, ssh2.shell, ssh2.exec, ssh2.tunnel, ssh2.scp, ssh2.sftp

**Reply by dvelopin** [reply]
This server is in the cloud on Amazon AWS within an Elastic Beanstalk EC2 instance

**Reply by dvelopin** [reply]
PHP Version 7.2.34. Yes have OpenSSL

**Reply by dvelopin** [reply]
Thank you for fixing my initial post, @finwe.

**Comment by dvelopin** [comment]
As it turns out allow URL fopen was turned off in the PHP configuration. Enabling this resolves the error.
